tools/flow_grid.py

```diff
@@ -0,0 +1,689 @@
+import numpy as np
+import pandas as pd
+import sys
+import ast
+
+class flow_grid(object):        
```

Comment:
Can we name this `FlowGrid`.  I like to use the pep8 CapWords for classes: https://www.python.org/dev/peps/pep-0008/#class-names